### PR TITLE
Fix doc levels and remove broken links

### PR DESCRIPTION
**Pull request by edenlightning**
## What does this PR do?

Fixes #\<issue_number>

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

### CODE REVIEW

**Review by Borda — APPROVED** [review]
seems that the git diff does not recognize correctly the renamed files...

**Comment by edenlightning** [comment]
@Borda i saw... any idea how to fix?

File: docs/source-pytorch/levels/advanced_level_15.rst [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
 
-################################################
-Level 15: Customize configs to run in production
-################################################
+###############################
+Level 15: Customize the trainer
+###############################
 
-This level goes over advanced YAML use for running models in production.
+In this level, you'll learn to modify the Trainer behavior.
```
Comment (lantiga) [inline docs/source-pytorch/levels/advanced_level_15.rst:7]:
```suggestion
In this level you'll learn to modify the behavior of the Trainer.
```

File: docs/source-pytorch/levels/advanced_level_16.rst [hunk 2]
```diff
@@ -1,10 +1,10 @@
 :orphan:
 
 ###############################
-Level 16: Customize the trainer
+Level 16: Own the training loop
 ###############################
 
-In this level, you'll learn to modify the Trainer behavior.
+Learn all the ways of owning your raw PyTorch loops with Lighting.
```
Comment (lantiga) [inline docs/source-pytorch/levels/advanced_level_16.rst:7]:
```suggestion
Learn all the ways of owning your raw PyTorch loops with Lightning.
```

File: docs/source-pytorch/levels/advanced_level_17.rst [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
 
-###############################
-Level 17: Own the training loop
-###############################
+#######################################
+Level 17: Enable advanced checkpointing
+#######################################
 
-Learn all the ways of owning your raw PyTorch loops with Lighting.
+This level shows you how to enable composable and/or cloud based checkpoints.
```
Comment (lantiga) [inline docs/source-pytorch/levels/advanced_level_17.rst:7]:
```suggestion
This level shows you how to enable composable and/or cloud-based checkpoints.
```

File: docs/source-pytorch/levels/intermediate_level_13.rst [hunk 4]
```diff
@@ -1,10 +1,10 @@
 :orphan:
 
 #################################
-Level 13: Optimize training speed
+Level 13: Run on on-prem clusters
 #################################
 
-In this level you'll use advanced profilers and mixed precision techniques to train bigger models faster.
+In this level you'll learn to run on onprem clusters.
```
Comment (lantiga) [inline docs/source-pytorch/levels/intermediate_level_13.rst:7]:
```suggestion
In this level you'll learn to run on on-prem clusters.
```

**Review by lantiga — COMMENTED** [review]
Added a few comments.

I have a fundamental question though: since you are changing level numberings, this will lead to pages changing URLs, which may also require Google to reindex, and broken links from external resources to the docs. Can't we just keep the name of the files as they are now?

**Comment by edenlightning** [comment]
@lantiga it might be confusing, but to avoid indexing problems i'll go with that suggestion- thanks!
 see https://github.com/Lightning-AI/lightning/pull/17545

**Pull request closed without merging**